py/dml/compat.py

```diff
@@ -118,6 +118,23 @@ class shared_logs_on_device(CompatFeature):
     last_api_version = api_6
 
 
+@feature
+class suppress_WLOGMIXUP(CompatFeature):
+    '''This compatibility feature makes it so the warning `WLOGMIXUP` is
+    suppressed by default. `WLOGMIXUP` warns about usages of a common faulty
+    pattern which results in broken log statements &mdash; for more
+    information, see the documentation of `WLOGMIXUP` in the
+    [Messages](messages.html) section of the DML 1.4 Reference Manual.
+
+    `WLOGMIXUP` is suppressed by default below Simics API version 7 in order
+    to avoid overwhelming users with warnings, as the faulty pattern that
+    `WLOGMIXUP` reports is very prevalent within existing code. Addressing
+    applications of the faulty pattern should be done before or as part of
+    migration to Simics API version 7.'''
```

Comment:
@MagnusIntel so your suggestion is to mention `--no-warn=WLOGMIXUP` here?

Reply:
I'm not sure. He could mean that it should mention that `--warn=WLOGMIXUP` can be used as an alternative to `--no-compat=suppress_WLOGMIXUP`. @MagnusIntel you'll have to clarify what you meant.

If so, it could be worth to also note there are some subtle differences between the two; `--warn=WLOGMIXUP` guarantees that the warning gets reported no matter what, while `--no-compat=suppress_WLOGMIXUP` just disables the _default_ suppression, but still allows for the warning to get suppressed explicitly via `--no-warn`.
I.e. `--warn=WLOGMIXUP --no-warn=WLOGMIXUP` will result in `WLOGMIXUP` being reported; in contrast, `--no-compat=suppress_WLOGMIXUP --no-warn=WLOGMIXUP` results in `WLOGMIXUP` being suppressed.

Reply:
I don't think we need to document these subtleties in detail, but there is a point in cross-referencing the corresponding --warn flag as an alternative. People will tweak flags until it works, and the migration logic will likely stay stable until it's deprecated out so whatever tweaks they do will continue to work.

Reply:
I think --warn=<TAG> is drastically easier to understand than --no-compat=suppress_<TAG>. And if I got the latter flags wrong, it just shows that the --no-compat flag is incomprehensible or unnecessarily hard to understand.

Reply:
So your complaint is rather that `--no-compat`, and compatibility features in general (like their naming scheme describing the legacy behavior rather than what the effect of disabling it is) is difficult to understand in general? 
I think both me and Erik agree... we've discussed previously about how we should present compatibility features, and if we should present them as "features" at all.
If a particular point of confusion is just the name of the `--no-compat` option, would it help if the option was called `--disable-compat` instead?

Regardless, addressing the presentation of `--no-compat` is orthogonal to this PR.

I can add in the documentation that `--warn=WLOGMIXUP` can serve as an alternative to `--no-compat=suppress_WLOGMIXUP`, but that's **only** applicable to `suppress_WLOGMIXUP`, as no other compatibility feature is about warnings. We are not going to add `--no-compat=suppress_<TAG>` in general (though that naming may become a pattern for future compatibility features, if we later want to introduce other warnings that we want to bind to API versions.)